File: tests/nodejs/cases/case_164_indirect_container_write_07.js
/**
 * Task case 164: indirect_container_write_07 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case164IndirectContainerWrite07(input) {
  const taskName = 'indirect_container_write_07';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const envelope = { wrapped: payload };
  // ESCAPE: payload stored via indirection into retained sink.
  retainedAudit.push(envelope.wrapped);
  return 'ok';
}

module.exports = {
  case164IndirectContainerWrite07,
};

File: tests/go/cases/case_164_indirect_container_write_07.go
package escape_tests

var retainedCase164 = []map[string]string{}

func Case164IndirectContainerWrite07(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "indirect_container_write_07",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "indirect_container_write_07:" + raw,
	}
	envelope := map[string]map[string]string{"wrapped": payload}
	// ESCAPE: indirection writes payload into retained sink.
	retainedCase164 = append(retainedCase164, envelope["wrapped"])
	return "ok"
}
